### ISSUE TITLE
NFT images are working but hidden image and json data is not showing up

### ISSUE THREAD

**Issue by logicdev406** [hashlips-lab/nft-erc721-collection#163]
Hi liarco,

I successfully deployed the NFT smart contract to rinkeby testnet and went over to opensea where the NFT showed up but the hidden image is not coming up I thought it was a problem with pinata because my NFT images and the hidden image are both hosted on pinata, but when I reveal the collection the NFT showed up so did the metadata of the nft I have cross checked the hidden image URL and redeploy the smart contract a few times but the hidden image is still not showing up pls I need help ASAP.

Thanks
Logic

**Comment by casuallycurious** [comment]
I've also have the same issue, I am going to restart the tutorial and see where I went wrong

**Comment by logicdev406** [comment]
I just don't know where the problem is coming from cause I checked the hidden metadata and image they are both hosted properly I then thought it was the smart contract not serving up the hidden metadata but when I checked the contract was serving up the hidden metadata now I just don't know where the problem is coming from

**Comment by BlueGreninja** [comment]
> I just don't know where the problem is coming from cause I checked the hidden metadata and image they are both hosted properly I then thought it was the smart contract not serving up the hidden metadata but when I checked the contract was serving up the hidden metadata now I just don't know where the problem is coming from

@logicdev406 @casuallycurious Have you guys tried refreshing the Metadata? After refreshing, give opensea some time to pull up the hidden images.

Also check if you added " ipfs//your_hidden_Metadata_CID_here/ " correctly

**Comment by logicdev406** [comment]
Yeah I check the link it's working and I have deployed about 4 of the smart contract still not working also tried refreshing the metadata but it didn't work.
I am just going to keep trying.

**Comment by logicdev406** [comment]
So if anyone is still having this issue I solve it and the problem was my hidden image metadata when I checked it with open-sea validate endpoint then I got an error saying the JSON was invalid due to a comma that was not supposed to be there. So if u are still having this issue I have got a few tips for u:

1, Check that the hidden image and metadata are correctly hosted.
2, When creating the hidden metadata which is basically a JSON file, use a code editor (E.g Vscode) that way the code editor will let u know if there are any errors in your JSON.
3. After deploying to Ethereum testnet or mainet validate your JSON file just to be sure that it works.

And that's all u need to do. 🤞
